src/Components/test/E2ETest/Tests/BootResourceCachingTest.cs

```diff
@@ -57,7 +57,7 @@ public void CachesResourcesAfterFirstLoad()
             var subsequentResourcesRequested = GetAndClearRequestedPaths();
             Assert.NotEmpty(initialResourcesRequested.Where(path => path.EndsWith("/blazor.boot.json")));
             Assert.Empty(subsequentResourcesRequested.Where(path => path.EndsWith("/dotnet.wasm")));
-            Assert.NotEmpty(subsequentResourcesRequested.Where(path => path.EndsWith(".js")));
+            Assert.Empty(subsequentResourcesRequested.Where(path => path.EndsWith(".js")));
```

Comment:
Why did this change? I would expect it always to fetch `blazor.webassembly.js` regardless of the caching, so I don't get how this can be right.

Reply:
Hmm, I didn't see it being requested.

Ah, it's likely due to the removal of the `no-cache` header we put in `blazor.webassembly.js`. We can chat a bit more about this, but it shouldn't be necessary for us to include the no-cache header, the browser will do the right thing.

The browser already does the right thing. We can talk about this later if you want.

Reply:
That doesn’t sound valid. We need the browser to request this since its content may have changed. Or if not, how does the browser do the right thing? I would expect the no-cache header is needed.

Let’s not consider this resolved just yet :)

Reply:
Discussed offline, I'll send a follow-up PR to resolve this.